### PR TITLE
Generate code for type checking databindings.

### PR DESCRIPTION
**Pull request by rictic**
I had to fork because I don't have write access to this repo.

### CODE REVIEW

File: src/result.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+export type Result<V, E> = Success<V, E>| Failure<V, E>;
+
+export namespace Result {
```
Comment (TimvdLippe) [inline src/result.ts:17]: This is originally from the analyzer. Maybe it is worth its own (npm) package?

Comment (rictic) [reply src/result.ts:17]: Yeah I think so. Some of this API isn't super useful or idiomatic though, will need a bit of iteration I expect.

**Comment by aomarks** [comment]
@rictic You now have write access.

**Comment by aomarks** [comment]
@rictic Ok if we close this PR? I'm moving this repo into the monorepo. I feel like even if we went ahead with this, it would probably make more sense in a separate package within the monorepo now.

**Pull request closed without merging**